docs/operations.md

```diff
@@ -28,6 +28,138 @@ Clients may inspect this specification at run-time so to obey the various limits
+Job events can be useful for user interfaces (UIs) and other applications that demand near-instantaneous updates.
+
+#### Architecture
+
+The job events notification system leverages [server-sent events](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events) (SSE) for its implementation.
+Below is a high-level overview of how the communication flow operates:
+
+```txt
+ ┌────────┐                            ┌─────┐
+ │ Client │                            │ wfx │
+ └────────┘                            └─────┘
+     |                                    |
+     |        HTTP GET /jobs/events       |
+     |-----------------------------------►|
+     |                                    |
+     |                                    |
+     |           Event Loop               |
+ ┌───|────────────────────────────────────|───┐
+ │   |  [Content-Type: text/event-stream] |   │
+ │   |                                    |   │
+ │   |                                    |   │
+ │   |            Push Event              |   │
+ │   |◄-----------------------------------|   │
+ │   |                                    |   │
+ └───|────────────────────────────────────|───┘
+     |                                    |
+     ▼                                    ▼
+ ┌────────┐                            ┌─────┐
+ │ Client │                            │ wfx │
+ └────────┘                            └─────┘
+```
+
+1. The client initiates communication by sending an HTTP `GET` request to the `/jobs/events` endpoint. Clients may also
+   include optional [filter parameters](#filter-parameters) within the request.
+2. Upon receipt of the request, `wfx` sets the `Content-Type` header to `text/event-stream`.
+3. The server then initiates a stream of job events in the response body, allowing clients to receive instant updates.
+
+#### Event Format Specification
+
+The event stream looks as follows:
+
+```
+data: [...]
+id: 1
+
+data: [...]
+id: 2
+
+[...]
+```
+
+with a single event being structured as follows:
+
+```
+data: { "action": "<ACTION>", "job": <JOB> }
+id: <EVENT_ID>
+
+```
+
+**Note**: Each event is terminated by a pair of newline characters `\n\n` (as required by the SSE spec).
+
+The semantics of the individual fields is:
+
+- `<ACTION>` specifies the type of event that occurred. The valid actions are:
+  - `CREATE`: a new job has been created
+  - `DELETE`: an existing job has been deleted
+  - `ADD_TAGS`: tags were added to a job
+  - `DELETE_TAGS`: tags were removed from a job
+  - `UPDATE_STATUS`: job status has been updated
+  - `UPDATE_DEFINITION`: job definition has been updated
+- `<JOB>` is the JSON representation of the job associated with this event
+- `<EVENT_ID>`: a (unique) numeric identifier for the event, starting at 1 and incremented by 1 for each subsequent event; can be used to detect message loss on the client side.
```

Comment:
→ can be used by the client to detect message loss
Whether or not this can happen depends on the channel's reliability and the OS handling of this channel. Consider UDP where this may happen while not so much for TCP (depending on how the OS implements reaction to delayed transmission over this channel).

Reply:
Revised